src/MiniCover.HitServices/HitTestMethod.cs

```diff
@@ -76,7 +76,9 @@ public static IEnumerable<HitTestMethod> MergeDuplicates(IEnumerable<HitTestMeth
 
         public void Serialize(Stream stream)
         {
+            Dictionary<int, int> hits;
             lock (_lock)
+                hits = this.HitedInstructions.ToDictionary(a => a.Key, a => a.Value);
```

Comment:
Sorry, I prefer the way it was.

Now I get why you have that lock. It's to avoid issues with iterator. But when this method is called the dictionary shouldn't be modified anymore. If it is being modified I think we have an issue somewhere else.

I think it's important to protect against 2 HitTestMethod writing to same file concurrently.

What do you think about?
```
lock(_lock)
lock(stream)
using (
```